### ISSUE TITLE
`reportInvalidTypeVarUse` superfluous when using default values?

### ISSUE THREAD

**Issue by randolf-scholz** [microsoft/pyright#10649]
Code sample in [pyright playground](https://pyright-play.net/?code=CYUwZgBGD20NoA0IF4IEsB2AXAugCgEMAnAcwC4IAbNAZy0RxQgDpWBKCAWgD4rb6EOCq2YAoIA)

```python
def foo[X = int](arg: list[X] = ...) -> list[X]: ...
```

`pyright` complains that

```
warning: Type variable "X" may go unsolved if caller supplies no argument for parameter "arg"
    Provide an overload that specifies the return type when the argument is not supplied (reportInvalidTypeVarUse)
```

But that seems superfluous, given the default value. `reveal_type(foo())` produces `list[int]` as expected and not `list[Unknown]`.

I mean, I can see that maybe one still wants to get a warning here, but it feels more like a lint than an actual typing error.

I would like to silence such errors globally, but `reportInvalidTypeVarUse` also covers other "real" errors like incorrect type var scoping.

**Comment by erictraut** [comment]
Is this code found in a stub file? I ask because `...` isn't a valid default argument value if this code were in a regular source file. Regardless of whether this is in a stub, if you replace `...` with a valid default arg value, you won't see this diagnostic.

It may be reasonable to suppress this check if the type variable has a default. I'll need to give that more thought. I added this check initially because I observed that many developers were confused about how type variables worked and were using them in a manner that didn't make sense. The goal with this diagnostic was to nudge them onto the correct path. So yes, this has more of a "lint" feel to it, but it's specifically targeted at (newish) users of the type system.

**Comment by randolf-scholz** [comment]
I usually put `= ...` for default values in overloads, where this also occurs.

**Comment by jorenham** [comment]
> I added this check initially because I observed that many developers were confused about how type variables worked and were using them in a manner that didn't make sense.

Yea this happens quite a lot, see e.g. https://github.com/numpy/numpy/pull/29278. But if the type parameter has an explicit default, then there shouldn't be problem I suppose.
